### ISSUE TITLE
Tests record verdicts instead of asserting them

### ISSUE THREAD

**Issue by EmilHvitfeldt** [tidymodels/tidypredict#311]
This is the reason a recent audit found 27 confirmed silent-wrong-prediction bugs against a clean `R CMD check` and 1770 passing tests, so it is worth treating as a first-class issue rather than a cleanup.

**The dominant test idiom does not assert correctness.** `expect_snapshot(tidypredict_test(model))` *records* the verdict. If a regression flips "All results are within the difference threshold" into an alert, that is an acceptable-looking snapshot diff, not a failure. The second most common idiom, `expect_identical(tidypredict_fit(model), tidypredict_fit(parse_model(model)))`, is a serialization round-trip: it proves the two paths agree with each other, not that either agrees with the model.

Files whose only correctness evidence is a snapshot:

| file                   | evidence                                                                                      |
|------------------------|-----------------------------------------------------------------------------------------------|
| `test-model-glm.R`     | 9 snapshots, **zero** `predict()` calls (lines 39-104)                                        |
| `test-model-xgboost.R` | 1521 lines, 48 `tidypredict_test()` calls, **3** `predict()` calls (`:747`, `:1371`, `:1382`) |
| `test-model-ranger.R`  | a single `predict()` comparison at `:47` covering 9 snapshots                                 |
| `test-model-rf.R`      | 5 `predict()` calls in 301 lines                                                              |
| `test-model-glmnet.R`  | 5 `predict()` calls in 428 lines; family and multinomial coverage is snapshot-only            |

`test-model-catboost.R` is the counter-example to copy: it calls `catboost.predict()` extensively from `:531`.

**Proposed change.** Every model test should include at least one assertion of the form

```r
expect_equal(
  rlang::eval_tidy(tidypredict_fit(model), df),
  unname(predict(model, df)),
  tolerance = 1e-12
)
```

with the tolerance loosened only where the engine genuinely computes in float32 (and a comment saying so). Where a snapshot is kept for its message, pair it with `expect_false(result$alert)` — `test-h2o.R:15,33` already does this.

**A second pattern worth fixing at the same time: the tests systematically exercise only the configuration that happens to work.** This is not one file's oversight, it recurs:

- every categorical catboost test uses exactly 3 factor levels, and 3 is essentially the only count that works;
- every ksvm test uses 2 numeric predictors, and the scaling bug needs exactly 1;
- every LightGBM NA test trains *with* NA present, the one `missing_type` where the routing is correct;
- the bart tests use `ntree = 1..10` where the dbarts default is 200;
- no rf/ranger test uses a factor predictor, which is where both are badly broken.

Adding these specific configurations would have caught most of the audit's findings.

Smaller related items:

- `test-model-glm.R:25` and `test-model-earth.R:21` mutate `model$coefficients <- round(..., 7)` before comparing, so both sides derive from the same mutated object. Label these "round trip" so they are not mistaken for correctness tests.
- Zero uses of `withr::local_seed()`; ~250 global `set.seed()` calls leak RNG state into later tests (`test-model-rf.R` reseeds 17 times). One genuinely unseeded random fit at `test-model-nnet.R:212`.
- `tests/testthat/helper-printing.R` `round_print()` has no callers and is fragile: `:10-12` passes a captured number to `sub()` as an unanchored regex, so `.` matches any character and a shorter number matches inside a longer one, and the pattern `[0-9]+\.?[0-9]+` never matches single-digit or `1e-05` values. Delete it or fix and use it.
- Backwards-compat fixtures: the v3 set is well designed (`pm` + `newdata` + `expected`, with `expected` from the modelling package's own `predict()`), but no fixture records the producing package or R version; `generate-v3-fixtures.R:26` uses one `set.seed(123)` for five sequential fits, so reordering silently changes the rf/xgb fixtures; there is no generator for `cubist-v2-regression.rds` at all; the v2 fixtures store no `expected` (structure only); and `test-backwards-compat-v3.R:9` uses `tolerance = 1e-6` where the suite standard is `1e-12`.
- Bare `tempfile()` instead of `withr::local_tempfile()` at `test-model-earth.R:24`, `test-model-glm.R:28`, `test-model-lm.R:29`, `test-model-glmnet.R:23,198`, `test-model-nullmodel.R:66,75`.
